moabb/paradigms/base.py

```diff
@@ -321,6 +331,38 @@ def get_data(  # noqa: C901
                             if len(self.filters) == 1
                             else mne.concatenate_epochs(proc)
                         )
+
+                        # prepare additional metadata
+                        if additional_metadata != "default":
+                            if not isinstance(dataset, BaseBIDSDataset):
+                                raise TypeError(
+                                    "Additional_metadata can only be used with BIDS datasets."
+                                )
+
+                            dm = load_bids_event_metadata(
+                                dataset, subject=subject, session=session, run=run
+                            )
```

Comment:
After discussion in the MOABB meeting, we think it would be useful in other datasets to have the option to get additional metadata columns (ex: ERPCore). So instead of having a special case for `BaseBIDSDataset`, the idea would be to have a method in `BaseDataset`:
```python
class BaseDataset:
    def get_additional_metadata(self, subject, session, run) -> None | pd.DataFrame:
        return None
```
that would be overwritten by the datasets that have additional metadata to pass

Reply:
The `load_bids_event_metadata` is currently using the `_find_matching_sidecar` from `mne_bids.path`. I was wondering if there is a data set which is `BaseDataset` but not `BaseBIDSDataset`, for which this approach might break. But happy to implement this in on the `BaseDataset` and always return potential additional meta data.